src/stores/RedisStore.ts

```diff
@@ -0,0 +1,33 @@
+import { connect } from "../../deps.ts";
+import type { Ratelimit } from "../types/types.d.ts";
+import { Store } from "./AbstractStore.ts";
+
+export class RedisStore extends Store {
+    private readonly store;
+
+    constructor(hostname?: string, port?: number) {
+        super();
+        this.store = connect({
+            hostname: hostname ?? "localhost",
+            port
+        })
+    }
+
+    public async get(ip: string) {
+        const data = await (await this.store).get(ip)
+        if (!data) return
+        return JSON.parse(data)
+    }
+
+    public async set(ip: string, ratelimit: Ratelimit) {
+        (await this.store).set(ip, JSON.stringify(ratelimit));
+    }
+
+    public async delete(ip: string) {
+        (await this.store).del(ip);
+    }
+
+    public async has(ip: string) {
+        return await (await this.store).exists(ip)
```

Comment:
Returning an `await` statement isn't considered the best practice. I would recommend defining a variable that `await` and then you return that variable instead.